### ISSUE TITLE
Analytics not working - which module should enable?

### ISSUE THREAD

**Issue by cptX** [Smile-SA/elasticsuite#2839]
Hi,
I have disabled the following modules to make elasticsuite lighter (I don't yet need the corresponding functionality of these modules):
Smile_ElasticsuiteAdminNotification
Smile_ElasticsuiteCatalogGraphQl
Smile_ElasticsuiteCatalogRule
Smile_ElasticsuiteCatalogOptimizer
Smile_ElasticsuiteTracker
Smile_ElasticsuiteThesaurus
Smile_ElasticsuiteSwatches
Smile_ElasticsuiteIndices
Smile_ElasticsuiteVirtualCategory

I have these modules enabled:

Smile_ElasticsuiteCore
Smile_ElasticsuiteCatalog
Smile_ElasticsuiteAnalytics

but I see nothing in the analytics in Elasticsuite menu.
![image](https://user-images.githubusercontent.com/9040042/217074158-f05a73fd-9f47-4114-95d0-7f2715837bcb.png)
So my question is what module should I enable to have the analytics functionality working?
I have the Smile_ElasticsuiteAnalytics enabled but this doesn't look enough...

Also is it possible in analytics to see the partial terms people typed before clicking on a suggested product?
In other words is it possible to monitor what the people typed for searching even without clicking the search button?

**Comment by romainruaud** [comment]
Hi, you need to have Smile_ElasticsuiteTracker.

That's the module responsible for collecting data. Analytics is just the "display" part.

You will see only search terms that are collected in the search result page, so not the partial ones that are typed in the autocomplete.

Regards

**Comment by cptX** [comment]
Please elaborate more, because your initial answer actually didn't gave a solution and doesn't justify the closure of the issue... Thanks!

**Comment by romainruaud** [comment]
When the ElasticsuiteTracker module is enabled, search terms are collected when this cronjob is run : 

https://github.com/Smile-SA/elasticsuite/blob/2.10.x/src/module-elasticsuite-tracker/Cron/IndexLogEvent.php

So, if you enabled Tracker and Analytics module, and have your Magento cron running, you should see data in the Analytics Dashboard.

Only data collected AFTER the enabling of the tracker, will be collected and displayed.

There is also one known issue with Fastly that requires a bit of custom configuration if you are using it as a CDN : https://github.com/Smile-SA/elasticsuite/wiki/FAQ#im-using-fastly-and-i-cant-see-anything-on-the-elasticsuite--search-usage-dashboard

**Comment by romainruaud** [comment]
In Analytics you can filter data by date, which is way more useful than just having the "raw" search terms.

You also see the conversion rate of each term, what has been spellchecked, the conversion rate of each term, and global metrics related to the search engine performance.

- Global metrics : number of searches, pages views, sessions with/without search and number of unique visitors.
- Popular search terms : to see what is actually trending, and how often a given search term is bringing conversion.
- Spellchecked search terms : to identify common mistakes and fix them with [Thesaurus](https://github.com/Smile-SA/elasticsuite/wiki/ThesaurusAdministration).
- 0 results search terms : to create a redirect for these search terms or to customize them.
- Percentage of sessions with search against sessions without search.
- Percentage of spellchecked requests.
- Conversion rate for session with search against sessions without search.

https://elastic-suite.github.io/documentation/docs/ElasticSuite/Analytics/Search%20Usage.html
